examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
+		frameFences[i] = device->createFence(video::IGPUFence::E_CREATE_FLAGS::ECF_SIGNALED_BIT);
+	}
+
+	// Record commands in commandBuffers here
+	for (uint32_t i = 0u; i < swapchainImageCount; ++i)
+	{
+		video::IGPUCommandBuffer::SImageMemoryBarrier undefToComputeTransitionBarrier;
+		undefToComputeTransitionBarrier.barrier.srcAccessMask = asset::EAF_TRANSFER_READ_BIT;
+		undefToComputeTransitionBarrier.barrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+		undefToComputeTransitionBarrier.oldLayout = asset::EIL_UNDEFINED;
+		undefToComputeTransitionBarrier.newLayout = asset::EIL_GENERAL;
+		undefToComputeTransitionBarrier.srcQueueFamilyIndex = ~0u;
+		undefToComputeTransitionBarrier.dstQueueFamilyIndex = ~0u;
+		undefToComputeTransitionBarrier.image = *(swapchainImages.begin() + i);
+		undefToComputeTransitionBarrier.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+		undefToComputeTransitionBarrier.subresourceRange.baseMipLevel = 0u;
+		undefToComputeTransitionBarrier.subresourceRange.levelCount = 1u;
+		undefToComputeTransitionBarrier.subresourceRange.baseArrayLayer = 0u;
+		undefToComputeTransitionBarrier.subresourceRange.layerCount = 1u;
+
+		video::IGPUCommandBuffer::SImageMemoryBarrier computeToPresentTransitionBarrier;
+		computeToPresentTransitionBarrier.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+		computeToPresentTransitionBarrier.barrier.dstAccessMask = static_cast<asset::E_ACCESS_FLAGS>(0);
+		computeToPresentTransitionBarrier.oldLayout = asset::EIL_GENERAL;
+		computeToPresentTransitionBarrier.newLayout = asset::EIL_PRESENT_SRC_KHR;
+		computeToPresentTransitionBarrier.srcQueueFamilyIndex = ~0u;
+		computeToPresentTransitionBarrier.dstQueueFamilyIndex = ~0u;
+		computeToPresentTransitionBarrier.image = *(swapchainImages.begin() + i);
+		computeToPresentTransitionBarrier.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+		computeToPresentTransitionBarrier.subresourceRange.baseMipLevel = 0u;
+		computeToPresentTransitionBarrier.subresourceRange.levelCount = 1u;
+		computeToPresentTransitionBarrier.subresourceRange.baseArrayLayer = 0u;
+		computeToPresentTransitionBarrier.subresourceRange.layerCount = 1u;
+
+		commandBuffers[i]->begin(0);
+
+		// Todo(achal): The fact that this pipeline barrier is solely on a compute queue might
+		// affect the srcStageMask. More precisely, I think, for some reason, that
+		// VK_PIPELINE_STAGE_TRANSFER_BIT shouldn't be specified in compute queue
+		// but present queue (or transfer queue if theres one??)
+		commandBuffers[i]->pipelineBarrier(asset::EPSF_TRANSFER_BIT,
+			asset::EPSF_COMPUTE_SHADER_BIT, 0, 0u, nullptr, 0u, nullptr, 1u,
+			&undefToComputeTransitionBarrier);
+
+		commandBuffers[i]->bindComputePipeline(pipeline.get());
+
+		const video::IGPUDescriptorSet* tmp[] = { descriptorSets[i].get() };
+		commandBuffers[i]->bindDescriptorSets(asset::EPBP_COMPUTE, pipelineLayout.get(),
+			0u, 1u, tmp);
+
+		commandBuffers[i]->dispatch(WIN_W, WIN_H, 1);
```

Comment:
why did you size your workgroups weirdly?

Reply:
just for easy testing, will remove in the future as I restore ex2 to original functionality

resolved

Reply:
not relevant anymore

resolved